src/updater.cc

```diff
@@ -42,18 +55,21 @@ PollingMetadataUpdater::~PollingMetadataUpdater() {
   }
 }
 
-void PollingMetadataUpdater::start() {
+bool PollingMetadataUpdater::ValidateConfiguration() const {
+  return period_ > seconds::zero();
```

Comment:
This should be `>=` -- `0` is a perfectly valid value (it means "don't poll")

Reply:
I think in the future when we use composition, this statement isn't true, but for what you're saying makes sense, updated.

Reply:
Depends on what you mean by "use composition". If it's in the context of "instead of inheritance", then that's orthogonal. If in the context of "composition of options", then you're probably right.
In any case, this looks good now.